src/views/darkroom.c

```diff
@@ -590,8 +590,11 @@ void expose(
                            darktable.lib->proxy.colorpicker.live_samples, FALSE);
   }
 
-  // draw guide lines if needed
-  if(!dev->gui_module || !(dev->gui_module->flags() & IOP_FLAGS_GUIDES_SPECIAL_DRAW))
+  const gboolean special_draw = dev->gui_module
+                             && dev->gui_module->flags() & IOP_FLAGS_GUIDES_SPECIAL_DRAW
+                             && dev->gui_module->enabled;
+  // draw guide lines if there is no active gui_module doing it via post_expose
+  if(!special_draw)
```

Comment:
So we do not call special draw if module is enabled now. Where it is called then now? Since this PR seems to work I'm missing something to understand how it works.

Reply:
Indeed. That would sometimes lead to grid drawing twice. :-)

Current code for all those modules have that flag assumes they do something whatever the enabled status is. So we have to make sure ashift paints the default.